app/models/concerns/event_concern.rb

```diff
@@ -3,6 +3,56 @@ module EventConcern
 
   included do
     after_destroy :raise_event_for_destroy, :prepend => true
+
+    def bulk_message
+      payload = case self.class.to_s
+                when "Endpoint", "Application"
+                  {
+                    :source                      => source,
+                    :applications                => source.applications,
+                    :endpoints                   => source.endpoints,
+                    :authentications             => authentications,
+                    :application_authentications => authentications.flat_map(&:application_authentications)
```

Comment:
@lindgrenj6 

it looks that those queries(and similar queries bellow) produces N+1 queries. (I am not sure how much it is problem regard to count of records)

maybe can be improved with just 2 queries: 

```
ApplicationAuthentication.includes(:authentication).where(:authentications => {:id => authentications.ids})
```

what do you think about it ?

Reply:
Great catch, I didn't think of that. I think I was able to replicate it with what I changed it to - rails automatically selects by id when passing in an array of AR objects. 

LMK what you think!

Reply:
I am sorry for delay, that's right and this looks good, thanks!